src/components/Sweep.tsx

```diff
@@ -0,0 +1,220 @@
+import { Box, Button, Divider, Flex, Skeleton, Stack, Text as CText } from '@chakra-ui/react'
+import { type AccountId, type AssetId, fromAssetId } from '@shapeshiftoss/caip'
+import type { UtxoBaseAdapter, UtxoChainId } from '@shapeshiftoss/chain-adapters'
+import { bn, FeeDataKey } from '@shapeshiftoss/chain-adapters'
+import type { Utxo } from '@shapeshiftoss/unchained-client/src/generated/bitcoin'
+import { useCallback, useEffect, useState } from 'react'
+import { useTranslate } from 'react-polyglot'
+import { Row } from 'components/Row/Row'
+import { getChainAdapterManager } from 'context/PluginProvider/chainAdapterSingleton'
+import { usePoll } from 'hooks/usePoll/usePoll'
+import { useWallet } from 'hooks/useWallet/useWallet'
+import { bnOrZero } from 'lib/bignumber/bignumber'
+import { useGetEstimatedFeesQuery } from 'pages/Lending/hooks/useGetEstimatedFeesQuery'
+import { selectAssetById } from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
+
+import { Amount } from './Amount/Amount'
+import { AssetIcon } from './AssetIcon'
+import { handleSend } from './Modals/Send/utils'
+
+const divider = <Divider />
+
+type SweepProps = {
+  assetId: AssetId
+  fromAddress: string | null
+  accountId: AccountId | undefined
+  onBack: () => void
+  onSweepSeen: () => void
+}
+
+const providerIcon = 'https://assets.coincap.io/assets/icons/rune@2x.png'
+
+export const Sweep = ({
+  assetId,
+  fromAddress,
+  accountId,
+  onBack: handleBack,
+  onSweepSeen: handleSwepSeen,
+}: SweepProps) => {
+  const [isSweepPending, setIsSweepPending] = useState(false)
+  const [txId, setTxId] = useState<string | null>(null)
+
+  const {
+    state: { wallet },
+  } = useWallet()
+  const translate = useTranslate()
+  const asset = useAppSelector(state => selectAssetById(state, assetId))
+
+  const { data: estimatedFeesData, isLoading: isEstimatedFeesDataLoading } =
+    useGetEstimatedFeesQuery({
```

Comment:
feels like it would be ideal to have these queries return the data expected, or throw so we can have an error payload to use for error modals/disabled states/etc. Right now we are using fallback values (ex `estimatedFeesData?.txFeeFiat ?? '0'`) which would be weird to see, and only trigger console.error on button press.

Reply:
Absolutely! `react-query` handles errors with the `isError, isSuccess, error` properties, so `  const estimatedFees = await estimateFees(estimateFeesInput)` within this query will reject and the error will be caught.
We don't really handle fees estimation errors at view-layer currently, which would be a nice addition but outside of this PR. But as far as the queries go, the support for throwing and handling errors is there, happy to do a quick producty sync after all lending PRs are in to see how we can handle errored state better.

In the meantime: 16adfd819c - which handles loading/error states properly across savers and lending consumooors

Reply:
:+1: makes sense to me and I guess the other part of what I was trying to say is it feels like changing `  if (!asset || !estimateFeesInput?.to || !estimateFeesInput.accountId) return` in the query function to throw instead of returning undefined so we don't have to have the undefined checks elsewhere and can just rely on the `isError` or `error` state. For follow up as you stated, but making sure that thought was translated correctly

Reply:
That's because the queryFn is extracted for use outside of react -since in react land, those *will* be actually defined with the enabled option, and outside of react may not, so that's effectively emualting the `enabled` option.
Not a fan of throwing, since this wouldn't affect react consumors (we would never hit `queryFn` because of the `enabled` option) but would mean try/catching when consuming `queryFn` directly as this is a regular async function.
Perhaps more boilerplaty but we could expect actually defined values and have consumors handle it to have a clean signature?

Reply:
I see what you are saying with the `enabled` versus raw usage. Well I won't get too hung up on this for now, but maybe we can take another look at some type cleanup after we get all of the functionality in.